src/components/imageOptionsEditor/imageOptionsEditor.js

```diff
@@ -78,35 +89,29 @@ define(['globalize', 'dom', 'dialogHelper', 'emby-checkbox', 'emby-select', 'emb
         });
     }
 
-    function editor() {
-        this.show = function (itemType, options, availableOptions) {
-            return new Promise(function (resolve, reject) {
-                var xhr = new XMLHttpRequest();
-                xhr.open('GET', 'components/imageOptionsEditor/imageOptionsEditor.template.html', true);
-
-                xhr.onload = function (e) {
-                    var template = this.response;
-                    var dlg = dialogHelper.createDialog({
-                        size: 'small',
-                        removeOnClose: true,
-                        scrollY: false
-                    });
-                    dlg.classList.add('formDialog');
-                    dlg.innerHTML = globalize.translateDocument(template);
-                    dlg.addEventListener('close', function () {
-                        saveValues(dlg, options);
-                    });
-                    loadValues(dlg, itemType, options, availableOptions);
-                    dialogHelper.open(dlg).then(resolve, resolve);
-                    dlg.querySelector('.btnCancel').addEventListener('click', function () {
-                        dialogHelper.close(dlg);
-                    });
-                };
-
-                xhr.send();
+export class showEditor {
+    constructor(itemType, options, availableOptions) {
+        return import('text!./components/imageOptionsEditor/imageOptionsEditor.template.html').then(({default: template}) => {
+            return new Promise((resolve) => {
+                const dlg = dialogHelper.createDialog({
+                    size: 'small',
+                    removeOnClose: true,
+                    scrollY: false
+                });
+                dlg.classList.add('formDialog');
+                dlg.innerHTML = globalize.translateDocument(template);
+                dlg.addEventListener('close', () => {
+                    saveValues(dlg, options);
+                });
+                loadValues(dlg, itemType, options, availableOptions);
+                dialogHelper.open(dlg).then(resolve, resolve);
+                dlg.querySelector('.btnCancel').addEventListener('click', () => {
+                    dialogHelper.close(dlg);
+                });
             });
-        };
+        });
     }
+}
 
-    return editor;
-});
+/* eslint-enable indent */
+export default showEditor;
```

Comment:
Classes should generally be in Capital Camel Case. Also try to avoid duplicate class names across files (Prefer a descriptive name like ImageOptionEditor to showEditor).